### PR TITLE
[fix bug 1458340] Add FB videochat funnelcake banner to /firefox/.

### PR DESCRIPTION
**Pull request by jpetto**
## Description

Adds a banner with a download button for funnelcake 134 (win/win64 only) to `/en-US/firefox/` for visitors matching the following:

- referring URL includes `facebook.com`
- en-US
- Windows, IE/Edge

## Issue / Bugzilla link

https://bugzilla.mozilla.org/show_bug.cgi?id=1458340
https://bugzilla.mozilla.org/show_bug.cgi?id=1452807

## Testing

Ensure UI looks good on IE 9+ and Edge.

https://www-demo4.allizom.org/en-US/firefox/

To activate organically, you must follow a link to the above from facebook.com using either Internet Explorer or Edge. I tested by creating a private post in FB with the link above, then clicking it using IE.

You can force the banner to appear with the following URL:

https://www-demo4.allizom.org/en-US/firefox/?test-banner=true

For both testing methods above, clicking the "close" X at the top right of the banner will prevent the banner from displaying again for 7 days. This being the case, I suggest testing in a private window (or manually clearing your cookies after dismissing).

### CODE REVIEW

**Comment by jpetto** [comment]
Currently blocked on GA, but code can be reviewed now.

**Comment by bensternthal** [comment]
Closes https://github.com/mozilla/bedrock/issues/5634

P.S. References are magical and we should use them!

**Comment by stephaniehobson** [comment]
Should I be able to test the funnelcake? I don't get facebook after downloading and installing.

**Comment by stephaniehobson** [comment]
Also, Firefox is predicting 1d and 57minutes to download the IE10 VM so... that's not happening today.

File: bedrock/firefox/templates/firefox/home.html [hunk 1]
```diff
@@ -19,10 +19,39 @@
...
+{% block site_header %}
+  {% if switch('firefox_home_fbvideochat_funnelcake', ['en-US']) %}
+    <div id="fbvideochat-banner" class="fbvideochat-banner">
+      <div class="content">
+        <div id="fbvideochat-close" class="fbvideochat-close"><button type="button"><span>Close</span></button></div>
+        <div class="header-content-wrapper">
+          <div class="header-content">
+            <p>Facebook Video Messenger calls are easy with</p>
+
+            {{ high_res_img('logos/firefox/logo-quantum-wordmark-large.png', {'alt': '', 'width': '185', 'height': '69', 'class': 'banner-logo'}) }}
```
[the remark this answers is not in the document]
Comment (jpetto) [reply bedrock/firefox/templates/firefox/home.html]: Ah, good catch.

**Comment by jpetto** [comment]
Added some fixes, and added an `.env` file for demo4 to allow it to serve the funnelcake. (I tested the install off demo4 and it did open `facebook.com` on first run.)

Also tested in IE 10 - had the same sticky nav issue as IE 9 - both should be fixed in this latest update.

**Comment by stephaniehobson** [comment]
IE10/Win7 ✅

**Comment by stephaniehobson** [comment]
IE9/Win7  ✅

**Comment by jpetto** [comment]
@stephaniehobson Hopefully you have time to review the final GA commit before your EOD. 😬

**Comment by jpetto** [comment]
@alexgibson I got this final commit in a bit late for Stephanie to review before her PTO - can you give it a look tomorrow?

**Comment by alexgibson** [comment]
> @alexgibson I got this final commit in a bit late for Stephanie to review before her PTO - can you give it a look tomorrow?

The GA changes lgtm! 

Not a blocker, but one non-GA related performance tweak here would be to only load the FB images when the banner is shown. Currently, every visitor get's to download them even if they are not on IE/Edge or being referred from FB. How long is this banner going to be around for?

**Comment by jpetto** [comment]
@alexgibson Good call on delaying image loading. Just added a commit to address.

The banner will be around until we get enough data to know if targeting this segment of visitors is worth it. If so, we'll likely keep it (or something similar) around long term.

**Pull request merged**